File: practice/algorithms/strings/beautiful_binary_string/BeautifulBinaryString.scala
object BeautifulBinaryString {

  def main(args: Array[String]) {
    val sc = new java.util.Scanner (System.in)
    val n = sc.nextInt()
    val B = sc.next()

    val beautiful = B.replaceAll("010", "b")
    val changes = beautiful.count(_ == 'b')
    println(changes)
  }
}

File: practice/algorithms/strings/beautiful_binary_string/BeautifulBinaryString.java
import java.util.*;

public class BeautifulBinaryString {

    public static void main(String[] args) {
        Scanner stdin = new Scanner(System.in);
        int l = Integer.parseInt(stdin.nextLine());
        String s = stdin.nextLine();
        String beautiful = s.replaceAll("010", "b");
        int changes = beautiful.replaceAll("[01]", "").length();
        System.out.println(changes);
        stdin.close();
    }
}